llvm/include/llvm/IR/Constants.h

```diff
@@ -98,6 +99,13 @@ class ConstantInt final : public ConstantData {
   /// value. Otherwise return a ConstantInt for the given value.
   static Constant *get(Type *Ty, uint64_t V, bool IsSigned = false);
 
+  /// WARNING: Incomplete support, do not use. These methods exist for early
+  /// prototyping, for most use cases ConstantInt::get() should be used.
+  /// Return a ConstantInt with a splat of the given value.
+  static ConstantInt *getSplat(LLVMContext &Context, ElementCount EC,
+                               const APInt &V);
+  static ConstantInt *getSplat(const VectorType *Ty, const APInt &V);
```

Comment:
I don't think these APIs should exist. `ConstantInt::get()` *already* supports creation of splats, they just aren't represented as `ConstantInt`s.

Reply:
We're in a transition period and thus I need an absolute way to create a vector ConstantInt (e.g. when parsing ll files and bitcode). Today `ConstantInt::get()` returns other `Constant` types to represent splats and that must be maintained for correctness because there are many code paths for which a vector ConstantInt will break.

Reply:
I think this isn't the right way to phase in the change. I think the `splat` syntax should just return whatever `ConstantVector::getSplat()` produces, and what that produces can be controlled by the opt flags you have introduced.

That means that the splat syntax becomes usable right away as a short-hand for producing the representations we currently use, and will switch to producing plain ConstantInt/ConstantFP once the flag is flipped (or in tests that explicitly flip it).

Reply:
OK, thanks.  It seems I've incorrectly assumed from an IR parsing and printing point of view there is a requirement for `IR_out == IR_in`. Your suggestion certainly means I can break the work up some more so I get that sorted.

Reply:
I've moved the textual IR side of things to https://github.com/llvm/llvm-project/pull/74620 following the suggestion to have `splat(x)` be synonymous with `ConstantInt/FP::get()`.